lib/Bio/EnsEMBL/DataCheck/Checks/FeaturePosition.pm

```diff
@@ -0,0 +1,108 @@
+=head1 LICENSE
+
+Copyright [2018] EMBL-European Bioinformatics Institute
+
+Licensed under the Apache License, Version 2.0 (the 'License');
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an 'AS IS' BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=cut
+
+package Bio::EnsEMBL::DataCheck::Checks::FeaturePosition;
+
+use warnings;
+use strict;
+
+use Moose;
+use Test::More;
+use Bio::EnsEMBL::DataCheck::Utils qw/sql_count/;
+
+extends 'Bio::EnsEMBL::DataCheck::DbCheck';
+
+use constant {
+    NAME        => 'FeaturePosition',
+    DESCRIPTION => 'Checks if features lie within bounds of seq_region i.e. start >=0 and end <= seq_region length.',
+    GROUPS      => [ 'corelike_handover' ],
+    DB_TYPES    => [ 'funcgen' ]
+};
+
+sub tests {
+  my ($self) = @_;
+
+  my $dna_dba = $self->get_dna_dba();
+  my $mapper = sub {
```

Comment:
You could do a core API call here, instead of the SQL, to get the list of seq_regions. E.g. 
`my $sa = $dna_dba->get_adaptor('Slice');`
`my $slices = $sa->fetch_all('toplevel');`
Then can iterate over the `$slices` array_ref, getting the required data from the `get_seq_region_id` and `seq_region_length` methods.

But... that could be quite slow. So SQL may be the best, but it could be simplified some. The `$region_name` is not used anywhere, so you could just retrieve seq_region_id and length, and do an `execute_into_hash` without the `$mapper` sub.

Reply:
Unfortunately `$region_name` is used to test if this is `regulatory_feature`, then the query is sligthly different.